src/register/data/selectors.js

```diff
@@ -1,33 +1,26 @@
-import { createSelector } from 'reselect';
-
 export const storeName = 'register';
 
-export const registerSelector = state => ({ ...state[storeName] });
-
-export const registrationErrorSelector = createSelector(
-  registerSelector,
-  register => register.registrationError.errorCode,
-);
-
-export const validationsSelector = createSelector(
-  registerSelector,
-  (register) => {
-    const { registrationError, validations } = register;
-
-    if (validations) {
-      return validations.validationDecisions;
-    }
-
-    if (Object.keys(registrationError).length > 0) {
-      const fields = Object.keys(registrationError).filter((fieldName) => !(fieldName in ['errorCode', 'usernameSuggestions']));
-
-      const validationDecisions = {};
-      fields.forEach(field => {
-        validationDecisions[field] = registrationError[field][0].userMessage || '';
-      });
-      return validationDecisions;
-    }
-
-    return null;
-  },
-);
+/**
+ * A helper for backend validations selector. It processes the api output and generates a
+ * key value dict for field errors.
+ * @param registrationError
+ * @param validations
+ * @returns {{username: string}|{name: string}|*|{}|null}
+ */
+export const getBackendValidations = (registrationError, validations) => {
```

Comment:
Does this need to be in the selector file? I think it can exist in the utils file and we can remove the registration selectors.